spring-rabbit/src/main/java/org/springframework/amqp/rabbit/config/AbstractRabbitListenerContainerFactory.java

```diff
@@ -194,8 +195,9 @@ public void setDefaultRequeueRejected(Boolean requeueRejected) {
 	 * @return the advice chain that was set. Defaults to {@code null}.
 	 * @since 1.7.4
 	 */
+	@Nullable
 	public Advice[] getAdviceChain() {
-		return Arrays.copyOf(this.adviceChain, this.adviceChain.length);
+		return this.adviceChain == null ? null : Arrays.copyOf(this.adviceChain, this.adviceChain.length);
```

Comment:
Why do we need copy here at all?
Since I see we do it anyway in the `setAdviceChain()`.

Looks like this is an issue only for the current `2.1.x` version

Reply:
Because the user could mutate the array but it won't take any effect because we built the proxy only during initialization.